lib/easypost/report.rb

```diff
@@ -0,0 +1,45 @@
+module EasyPost
+  class Report < Resource
+    def self.create(params={}, api_key=nil)
+      url = self.url
+      wrapped_params = {}
+      wrapped_params[self.class_name().to_sym] = params
+      report_types = ['shipment', 'payment_log', 'tracker']
+
+      if report_types.include?(params[:type].to_s)
+        url += "/#{params[:type]}"
+      else
+        raise Error.new("Undertermined Report Type")
+      end
+      response, api_key = EasyPost.request(:post, url, api_key, params)
+      return Util.convert_to_easypost_object(response, api_key)
+    end
+
+    def self.retrieve(params={}, api_key=nil)
+      url = self.url
+      report_types = { shprep: 'shipment', plrep: 'payment_log', trkrep: 'tracker' }
```

Comment:
This could be moved to the class level, as its keys are repeated for all of these functions.